build/scripts/tests.sh

```diff
@@ -27,29 +27,29 @@ need_publish=(
     'src/Compilers/CSharp/Test/Symbol/CSharpCompilerSymbolTest.csproj'
 )
 
-for project in $need_publish
+for project in "${need_publish[@]}"
 do
     echo "Publishing ${project}"
-    dotnet publish --no-restore ${root_path}/${project} -r ${runtime_id} -f ${target_framework} -p:SelfContained=true
+    dotnet publish --no-restore "${root_path}"/"${project}" -r "${runtime_id}" -f "${target_framework}" -p:SelfContained=true
 done
 
 # Discover and run the tests
 mkdir -p "${log_dir}"
 
-for test_path in ${unittest_dir}/*/${target_framework}
+for test_path in "${unittest_dir}"/*/"${target_framework}"
 do
-    publish_test_path=${test_path}/${runtime_id}/publish
-    if [ -d ${publish_test_path} ]
+    publish_test_path="${test_path}"/"${runtime_id}"/publish
+    if [ -d "${publish_test_path}" ]
     then
-        test_path=${publish_test_path}
+        test_path="${publish_test_path}"
     fi
 
-    file_name=( ${test_path}/*.UnitTests.dll )
-    log_file="${log_dir}/$(basename "${file_name%.*}.xml")"
-    deps_json="${file_name%.*}.deps.json"
-    runtimeconfig_json="${file_name%.*}.runtimeconfig.json"
-    echo "Running ${file_name}"
-    dotnet exec --depsfile "${deps_json}" --runtimeconfig "${runtimeconfig_json}" ${xunit_console} "$file_name" -xml "${log_file}"
+    file_name=( "${test_path}"/*.UnitTests.dll )
+    log_file="${log_dir}"/"$(basename "${file_name%.*}.xml")"
+    deps_json="${file_name%.*}".deps.json
+    runtimeconfig_json="${file_name%.*}".runtimeconfig.json
+    echo Running "${file_name[@]}"
+    dotnet exec --depsfile "${deps_json}" --runtimeconfig "${runtimeconfig_json}" "${xunit_console}" "${file_name[@]}" -xml "${log_file}"
```

Comment:
This actually might be incorrect... I'm not sure if xunit supports passing multiple dlls in a single run. If there _are_ multiple dlls, though, something is horribly broken, and failing is okay.

Reply:
I wasn't sure about this one, but your reasoning sounds right